src/cls/rgw/cls_rgw_ops.h

```diff
@@ -1002,56 +1002,86 @@ struct cls_rgw_lc_get_next_entry_op {
 WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_op)
 
 struct cls_rgw_lc_get_next_entry_ret {
-  pair<string, int> entry;
+private:
+  mutable pair<string, int> compat_entry; // only for backward compatibility
+public:
+  cls_rgw_lc_entry entry;
 
   cls_rgw_lc_get_next_entry_ret() {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 1, bl);
+    compat_entry = make_pair(entry.bucket, entry.status);
```

Comment:
You shouldn't need to encode the compat entry, the versioning takes care of that.  Only the decode needs to handle it.  This goes for all the encodes.